<a href="https://colab.research.google.com/github/renanpbento/python-training/blob/main/IGTI_Mod_1_Challenge.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**IGTI Bootcamp - Module I: Challenge**

Goals - practice the concepts presented in the module:
- Loops
- Range
- Functions
- Installing and importing modules/libraries

Creating functions for the exercise

In [ ]:
# Function 1 (def)

def saudacao(nome):
  print('Olá', nome, 'tudo bem?')

In [ ]:
# Testing Function 1

saudacao('Roberto')

Olá Roberto ,  tudo bem?


In [ ]:
# Function 2 (def)

def imc(peso, altura):
  print(peso/(altura**2))

In [ ]:
# Testing Function 2

imc(80, 1.65)

29.384756657483933


In [ ]:
# Function 3 (lambda)

soma = lambda x, y: x + y

In [ ]:
# Testing Function 3

soma(3,4)

7

In [ ]:
# Function 4 (def)

def soma(x, y):
  print(x + y)

In [ ]:
# Testing Function 4

soma(2,2)

4


In [ ]:
# Loop 1 (while)

num = 0
while num < 20:
  print(num)
  num +=1

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19


In [ ]:
# Loop 2 (for)

for i in range(0, 10, 2):
  print(i)

0
2
4
6
8


In [ ]:
# Loop 3 (for, if)

numeros = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

for x in numeros:
  print(x)
  if x == 3:
    break

1
2
3


In [ ]:
# Loop 4 (while, if)

num = 0

while num < 10:
  num += 1
  if num == 6:
    continue
  print(num)

1
2
3
4
5
7
8
9
10




---

**Answers to the challenge's questions**

**Q1: O que a função 1 faz?** 

(d) Imprime uma mensagem de saudação, variando de acordo com o nome inserido no argumento da função.


**Q2: O que a função 2 faz?**



(a) Calcula o IMC dividindo o peso pela altura e elevando a 2.

**Q3: O que a função 3 faz?**


(a) Soma entre dois valores.

**Q4: Após executar a função 1, caso eu escreva saudacao(‘Roberto’), qual será o resultado?**

(c) Olá Roberto tudo bem?

**Q5: Após executar a função 2, caso eu escreva imc(80, 1.65), qual será o resultado?**

(a) 29.384756657483933

**Q6: Na função 3, o que representa as letras x, y?**

(d) São os argumentos da função.

**Q7: Há um erro na função 4? Se sim, qual?**

(a) Sim. Falta um sinal de dois pontos na primeira linha.

**Q8: No laço de repetição 1, qual será o último número a aparecer?**

(c) 19

**Q9: No laço de repetição 1, caso eu apague o comando num += 1, o que acontecerá?**

(c) Um loop infinito

**Q10: Qual é o último número que vai aparecer no laço de repetição 2?**

(d) 8

**Q11:  O que faz o break no laço de repetição 3?**

(a) Para de executar o loop quando o x chegar no número 3.

**Q12:  O que faz o continue no laço de repetição 4?**

(a) Continua a execução do loop omitindo (pulando) o número 6.

**Q13: O que são módulos em python?**

(d) São um conjunto de funções e códigos que podem ser importados, instalados ou criados pelo desenvolvedor.

**Q14: Qual o nome do gerenciador de pacotes em python? Usado para instalar e desinstalar pacotes em python.**

(a) pip

**Q15:  Qual o comando que devemos utilizar para importar um pacote?**

(a) import

In [ ]:
# An alternative approach using dictionary:

mod_1_challenge = {1:'d',
                   2:'a',
                   3:'a',
                   4:'c',
                   5:'a',
                   6:'d',
                   7:'a',
                   8:'c',
                   9:'c',
                   10:'d',
                   11:'a',
                   12:'a',
                   13:'d',
                   14:'a',
                   15:'a'}

print(mod_1_challenge)

{1: 'd', 2: 'a', 3: 'a', 4: 'c', 5: 'a', 6: 'd', 7: 'a', 8: 'c', 9: 'c', 10: 'd', 11: 'a', 12: 'a', 13: 'd', 14: 'a', 15: 'a'}


Author: Renan Pereira Bento